tests/test_conf.py

```diff
@@ -69,3 +69,18 @@ def test_parser_read_without_header(self):
         read_file_method = getattr(parser, 'read_file', parser.readfp)
         read_file_method(StringIO('foo: bar'))
         self.assertEqual(parser.get('general', 'foo'), 'bar')
+
+    def test_file_permission_warning(self):
+        """Warn file permissions may expose credentials
+        """
+        with mock.patch.object(warnings, 'warn') as warn:
+            with mock.patch.object(os.path, 'isfile') as isfile:
+                with mock.patch.object(os, 'stat') as os_stat:
+                    isfile.return_value = True
+                    import posix
+                    test_statz = posix.stat_result([0o604,0,0,0,0,0,0,0,0,0])
```

Comment:
This is pretty gnarly looking, you can probably use the bitmask here too to make it a bit cleaner without using the posix module.